crates/ollama/src/ollama.rs

```diff
@@ -70,11 +70,36 @@ pub struct Model {
     pub keep_alive: Option<KeepAlive>,
 }
 
+// This could be dynamically retrieved via the API (1 call per model)
+// curl -s http://localhost:11434/api/show -d '{"model": "llama3.1:latest"}' | jq '.model_info."llama.context_length"'
+fn get_max_tokens(name: &str) -> usize {
+    match name {
+        "dolphin-llama3:8b-256k" => 262144, // 256K
+        _ => match name.split(':').next().unwrap() {
+            "mistral-nemo" => 1024000,                                      // 1M
+            "deepseek-coder-v2" => 163840,                                  // 160K
+            "llama3.1" | "phi3" | "command-r" | "command-r-plus" => 131072, // 128K
+            "codeqwen" => 65536,                                            // 64K
+            "mistral" | "mistral-large" | "dolphin-mistral" | "codestral"   // 32K
+            | "mistral-openorca" | "dolphin-mixtral" | "mixstral" | "llava"
+            | "qwen" | "qwen2" | "wizardlm2" | "wizard-math" => 32768,
+            "codellama" | "stable-code" | "deepseek-coder" | "starcoder2"   // 16K
+            | "wizardcoder" => 16384,
+            "llama3" | "gemma2" | "gemma" | "codegemma" | "dolphin-llama3"  // 8K
+            | "llava-llama3" | "starcoder" | "openchat" | "aya" => 8192,
+            "llama2" | "yi" | "llama2-chinese" | "vicuna" | "nous-hermes2"  // 4K
+            | "stablelm2" => 4096,
+            "phi" | "orca-mini" | "tinyllama" | "granite-code" => 2048,     // 2K
+            _ => 2048,                                                      // 2K (default)
+        },
```

Comment:
Rad, I think this is great.

Reply:
It's going to be slightly wrong still as the tokenization is different between many of these models and tiktoken itself. However, it's a good estimate for the user.

Reply:
The estimates are still fuzzy, but previously our requests was including `num_ctx` 2048 which caused ollama to prematurely truncate context there.

Reply:
Oh yikes, I didn't realize this was truncating it! I thought it was just our estimate for the window. Thank you!

Reply:
Yep.
```sh
# tail -f ~/.ollama/logs/server.log
INFO [update_slots] input truncated | n_ctx=2048 n_erase=74657 n_keep=4 n_left=2044 n_shift=1022 tid="0x1f2dccf40" timestamp=1724689205
```